### PR TITLE
LPS-180756  Add Object Entry Performance Test

### PR DESCRIPTION
**Pull request by mateussgn**
Related issue: https://issues.liferay.com/browse/LPS-180756

### CODE REVIEW

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryPerformanceTest.java [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+				"ms"));
+	}
+
+	private void _writeToLogFile(String... contents) throws IOException {
+		System.out.println(Arrays.toString(contents));
+
+		Files.write(
+			_logFilePath, Arrays.asList(contents), StandardOpenOption.APPEND,
+			StandardOpenOption.CREATE, StandardOpenOption.WRITE);
+	}
```
Comment (carolmariaabb) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryPerformanceTest.java:187]: @mateussgn all of these methods are only used once. Can we simplify to:

```
private Closeable _timeSpent() {
	Thread thread = Thread.currentThread();

	StackTraceElement stackTraceElement = thread.getStackTrace()[3];

	String invokerName = StringBundler.concat(
		stackTraceElement.getClassName(), StringPool.POUND,
		stackTraceElement.getMethodName());

	long startTime = System.currentTimeMillis();

	return () -> Files.write(
		_logFilePath,
		Arrays.asList(
			StringBundler.concat(
				invokerName, " used ",
				System.currentTimeMillis() - startTime, "ms")),
		StandardOpenOption.APPEND, StandardOpenOption.CREATE,
		StandardOpenOption.WRITE);
}
```
?

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryPerformanceTest.java [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+		long startTime = System.currentTimeMillis();
+
+		return () -> _writeToLogFile(
+			StringBundler.concat(
+				invokerName, " used ", System.currentTimeMillis() - startTime,
+				"ms"));
+	}
+
+	private void _writeToLogFile(String... contents) throws IOException {
+		System.out.println(Arrays.toString(contents));
```
[the remark this answers is not in the document]
Comment (mateussgn) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryPerformanceTest.java:182]: I forgot to delete it, I was doing tests with this line. Sure, I'll delete it.

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryPerformanceTest.java [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+	public void testPerformanceGetObjectEntries() throws Exception {
+		try (Closeable closeable = _timeSpent()) {
+			_objectEntries = _objectEntryLocalService.getObjectEntries(
+				0, _objectDefinition.getObjectDefinitionId(), QueryUtil.ALL_POS,
+				QueryUtil.ALL_POS);
+		}
+	}
+
+	private void _addObjectEntries() throws Exception {
+		for (int counter = 0; counter < 1000; counter++) {
```
[the remark this answers is not in the document]
Comment (mateussgn) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryPerformanceTest.java:146]: Sure! Great idea.

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryPerformanceTest.java [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+
+		_objectDefinition =
+			_objectDefinitionLocalService.publishCustomObjectDefinition(
+				TestPropsValues.getUserId(),
+				_objectDefinition.getObjectDefinitionId());
+
+		_objectEntryManager = _objectEntryManagerRegistry.getObjectEntryManager(
+			_objectDefinition.getStorageType());
+
+		_adminUser = TestPropsValues.getUser();
```
[the remark this answers is not in the document]
Comment (mateussgn) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryPerformanceTest.java:113]: I'll let it as it is, so I can use the `_adminUser` inside `_addObjectEntries` in the `new DefaultDTOConverterContext` call

**Comment by mateussgn** [comment]
ci:test:sf

**Comment by mateussgn** [comment]
ci:test:relevant

**Review by carolmariaabb — APPROVED** [review]
LGTM! 🎉

**Comment by mateussgn** [comment]
ci:test:sf

**Comment by mateussgn** [comment]
ci:test:object

**Comment by mateussgn** [comment]
ci:test:relevant

**Comment by guilhermedcamacho** [comment]
See #2061

**Pull request closed without merging**